isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+#include "Isis.h"
+
+#include "Application.h"
+#include "Cube.h"
+#include "CubeAttribute.h"
+#include "FileName.h"
+#include "Portal.h"
+#include "ProcessByLine.h"
+#include "Pvl.h"
+#include "SpecialPixel.h"
+#include "UserInterface.h"
+
+#include <QString>
+
+using namespace std;
+using namespace Isis;
+
+// Taken from ProcessMosaic
+const int FLOAT_MIN = -16777215;
+const int FLOAT_MAX = 16777216;
+
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand);
+void createMosaicCube(UserInterface ui, QString bands);
+void createTrackCube(UserInterface ui, QString trackingBand);
+void copyPixels(Buffer &in, Buffer &out);
+void copyTrackPixels(Buffer &in, Buffer &out);
+
+void IsisMain() {
+  UserInterface &ui = Application::GetUserInterface();
+  QString copyBands;
+  QString trackBand;
+  findTrackBand(ui, copyBands, trackBand);
+  createMosaicCube(ui, copyBands);
+  createTrackCube(ui, trackBand);
+}
+
+
+/**
+ * Finds the index of the tracking band in the BandBin group and stores the index
+ * in trackBand. The other bands' indices are stored in copyBands.
+ * We store the indices in QStrings because we use them as cube attributes when
+ * we process the mosaic and tracking cubes.
+ *
+ * @param ui        The user interface
+ * @param copyBands Indices of the bands to copy over to the mosaic cube
+ * @param trackBand Index of the tracking band
+ */
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand) {
+  Cube inputCube = Cube(ui.GetFileName("FROM"));
+  Pvl *inputLabel = inputCube.label();
+
+  if (inputLabel->hasObject("IsisCube")) {
+    PvlObject &cubeObject = inputLabel->findObject("IsisCube");
+    if (cubeObject.hasGroup("BandBin")) {
+      PvlGroup &bandBinGroup = cubeObject.findGroup("BandBin");
+      PvlKeyword &currentKeyword = bandBinGroup[0];
```

Comment:
This will error if the BandBin group has no keywords. I doubt that's something that will happen normally, but you may want to handle that scenario with a more descriptive error message than the PvlGroup index out of bands message.